In [1]:
import pandas as pd
import numpy as np
import tensorflow as tf
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline 
sns.set(color_codes=True)

Tablica koja prikazuje prosječni score svake osobe za svaku vještinu i koju strategiju je ta osoba odabrala.

In [2]:
df = pd.read_excel("SEVER_2.xlsx")
df.iloc[:,0] = df.iloc[:,0]/10
df.iloc[:,1] = df.iloc[:,1]/10
df.iloc[:,2] = df.iloc[:,2]/10
df.head(5)

Vjestina aktivnog slusanja  Vjestina umrezavanja  \
0                         4.6                   4.0   
1                         4.2                   3.6   
2                         4.8                   4.2   
3                         4.8                   4.8   
4                         4.0                   5.0   

   Vjestina utjecaja na druge  Tvrdo pregovaranje  Pozicijsko pregovaranje  \
0                         3.6                   0                        0   
1                         5.0                   0                        0   
2                         4.4                   0                        1   
3                         4.2                   0                        0   
4                         4.6                   1                        0   

   Meko pregovaranje  Fiktivno pregovaranje  Principijelno pregovaranje  Spol  
0                  0                      0                           1     1  
1                  0                      0                           1     1  
2                  0                      0                           0     1  
3                  0                      0                           1     1  
4                  0                      0                           0     1

In [3]:
# Create an empty list
filter_arr = []
# go through each element in arr
for i in range(0, 30):
  if 1 == df.iloc[i,8]:
    filter_arr.append(True)
  else:
    filter_arr.append(False)

In [4]:
vas_zene = df.iloc[:,0][filter_arr].to_numpy()

In [5]:
vas_zene

array([4.6, 4.2, 4.8, 4.8, 4. , 4. , 4.4, 4. , 4. , 3.6, 3.6, 4.4, 4.8,
       4.4, 4. , 4.4])

In [6]:
filter_arr_1 = np.invert(filter_arr)

In [7]:
filter_arr_1

array([False, False, False, False, False, False,  True, False,  True,
       False,  True, False,  True, False,  True,  True, False, False,
        True,  True, False,  True,  True,  True, False,  True, False,
        True,  True, False])

In [8]:
vas_muskarci = df.iloc[:,0][filter_arr_1].to_numpy()

In [9]:
np.mean(vas_zene)

4.25

Prosjek za vještinu aktivnog slušanja za žene je 4.25.

In [10]:
np.mean(vas_muskarci)

4.185714285714286

Prosjek za vještinu aktivnog slušanja za osobe koje su odabrale druge strategije je 4.1857143. 

Muškarci imaju nešto malo manju vještinu aktivnog slušanja.

In [11]:
vu_zene = df.iloc[:,1][filter_arr].to_numpy()

In [12]:
vu_muskarci = df.iloc[:,1][filter_arr_1].to_numpy()

In [13]:
np.mean(vu_zene)

4.2250000000000005

Prosjek za vjestinu umrezavanja za zene je 4.225

In [14]:
np.mean(vu_muskarci)

4.1000000000000005

Prosjek za vjestinu umrezavanja za muskarce je 4.1.

Dakle, muskarci imaju malo manje razvijenu vjestinu umrezavanja.

In [15]:
vud_zene = df.iloc[:,2][filter_arr].to_numpy()
vud_muskarci = df.iloc[:,2][filter_arr_1].to_numpy()

In [16]:
np.mean(vud_zene)

4.225

Prosjek za vjestinu utjecaja na druge za zene je 4.225.

In [17]:
np.mean(vud_muskarci)

4.385714285714286

Prosjek za vjestinu utjecaja na druge za muskarce je 4.3857...

Muskarci imaju malo razvijeniju vjestinu utjecaja na druge osobe.

## Whitney Mann U Test

Whitney Mann U test ima nul-hipotezu H0: distribucija jedne populacije je jednaka distribuciji druge populacije.

Prvo ćemo usporediti distribuciju vjestine aktivnog slusanja izmedju zena i muskaraca.

In [18]:
# Uzorak osoba za zene
vas_zene

array([4.6, 4.2, 4.8, 4.8, 4. , 4. , 4.4, 4. , 4. , 3.6, 3.6, 4.4, 4.8,
       4.4, 4. , 4.4])

In [19]:
# Uzorak za muskarce
vas_muskarci

array([3.4, 3.6, 4.6, 5. , 3.8, 4. , 4.6, 4.4, 4.4, 4.4, 3.6, 4.2, 4. ,
       4.6])

In [20]:
import scipy
scipy.stats.mannwhitneyu(vas_zene, vas_muskarci)

MannwhitneyuResult(statistic=120.0, pvalue=0.7515154038124412)

Jako velika p-vrijednost (cca 75%), dakle ne odbacujemo H0 hipotezu, odnosno ne odbacujemo hipotezu koja kaze da su distribucije tih dviju populacija jednake.

Odnosno, ne odbacujemo hipotezu koja kaze da su vjestine aktivnog slusanja kod zena jednake kao i kod muskaraca.

Sada na isti nacin ispitamo za vjestinu umrezavanja

H0: vjestine umrezavanja za zene imaju jednaku distribuciju kao i vjestine umrezavanja muskaraca

In [21]:
import scipy
scipy.stats.mannwhitneyu(vu_zene, vu_muskarci)

MannwhitneyuResult(statistic=124.0, pvalue=0.6291133696243867)

Velika p-vrijednost (cca 62.9%), dakle ne odbacujemo H0 hipotezu.

I na isti nacin ispitamo i za trecu vjestinu.

In [22]:
import scipy
scipy.stats.mannwhitneyu(vud_zene, vud_muskarci)

MannwhitneyuResult(statistic=86.0, pvalue=0.28304350077716733)

Velika p-vrijednost (cca 28.3%), dakle ne odbacujemo H0.